lib/Alchemy/Phrasea/Controller/Prod/ToolsController.php

```diff
@@ -251,6 +251,35 @@ public function applyThumbnailExtractionAction(Request $request)
         try {
             $record = new \record_adapter($this->app, $request->request->get('sbas_id'), $request->request->get('record_id'));
 
+            $subDef = $request->request->get('sub_def');
+
+            if( is_array($subDef)) {
+                foreach($subDef as $def) {
+                    $subDefName = $def['name'];
+                    $subDefDataUri = $def['src'];
+
+                    $dataUri = Parser::parse($subDefDataUri);
```

Comment:
Extract following lines in its own private method. It can also be used for legacy code.